File: BudgetApp.Client/src/app/entities/Budget.ts
import { Expense } from "./Expense";
import { IncomeSource } from "./IncomeSource";

export class Budget {
    budgetId: string;
    expenses: Expense[];
    incomeSources: IncomeSource[];

  constructor(budgetId: string, expenses: Expense[], incomeSources: IncomeSource[]) {
    this.budgetId = budgetId;
    this.expenses = expenses;
    this.incomeSources = incomeSources;
  }
}

File: BudgetApp.Domain/Entities/Budget.cs
namespace BudgetApp.Domain.Entities
{
    public class Budget
    {
        public Guid BudgetId { get; set; }
        public virtual List<Expense>? Expenses { get; set; } = new List<Expense>();
        public virtual List<IncomeSource>? IncomeSources { get; set; } = new List<IncomeSource>();
    }
}
